PsfLauncher/main.cpp

```diff
@@ -550,54 +618,84 @@ ErrorInformation CheckIfPowershellIsInstalled(bool& isPowershellInstalled) noexc
     return {};
 }
 
+ErrorInformation CheckIfShouldRun(bool shouldRunOnce, bool &canScriptRun)
+{
+    canScriptRun = true;
+    if (shouldRunOnce)
+    {
+        std::wstring runOnceSubKey = L"SOFTWARE\\";
+        runOnceSubKey.append(psf::current_package_full_name());
+        runOnceSubKey.append(L"\\runOnce");
+
+        DWORD keyDisposition;
+        wil::unique_hkey registryHandle;
+        LSTATUS createResult = RegCreateKeyExW(HKEY_CURRENT_USER, runOnceSubKey.c_str(), 0, nullptr, REG_OPTION_NON_VOLATILE, KEY_ALL_ACCESS, nullptr, &registryHandle, &keyDisposition);

```

Comment:
Since you are already writing a key, why not just have a boolean value here and check on that.

Reply:
What do you mean?